test/test_evaluation.jl

```diff
@@ -218,3 +218,39 @@ end
             basic_eval ≈ many_ops_eval
     end
 end
+
+@testitem "Disable early exit" begin
+    using DynamicExpressions
+    using Bumper, LoopVectorization
+
+    T = Float16
+    ex = @parse_expression(
+        2 * x, binary_operators = [*], variable_names = ["x"], node_type = Node{T}
+    )
+    X = T[1.0 floatmax(T)]
+    @test all(isnan.(ex(X)))
+    @test ex(X; options=EvaluationOptions(early_exit=Val(false))) ≈ [2.0, Inf]
+
+    for turbo in [Val(false), Val(true)],
+        T in [Float32, Float64],
+        bumper in [Val(false), Val(true)]
+
+        ex = @parse_expression(
+            (-b - sqrt(b^2 - (4 * a) * c)) / (2 * c),
+            binary_operators = [-, *, /, ^],
+            unary_operators = [-, sqrt],
+            variable_names = ["a", "b", "c"],
+            node_type = Node{T}
+        )
+        X = T[
+            -1 -1
+            1 floatmax(T)
+            1 1
+        ]
+        @test all(isnan.(ex(X; options=EvaluationOptions(bumper=bumper, turbo=turbo))))
+        y = ex(X; options=EvaluationOptions(bumper=bumper, turbo=turbo, early_exit=false))
+        @test y[1] == T(-1.618033988749895)
+        # FIXME: this is NaN on macOS and -Inf on windows/ubuntu...
+        @test !isfinite(y[2])
```

Comment:
Before starting with the `EvaluationOptions` this was the only test that failed. The test should pass on every OS like this now, but I am not sure if ignoring it being NaN on one OS/Inf on another is that nice. What do you think?

Reply:
It might just be an architecture-specific behavior of overflows, rather than an issue on the package's side?